src/main/kotlin/com/terraformation/backend/admin/AdminVotersController.kt

```diff
@@ -0,0 +1,251 @@
+import com.terraformation.backend.log.perClassLogger
+import org.jooq.DSLContext
+import org.jooq.Record
+import org.springframework.stereotype.Controller
+import org.springframework.ui.Model
+import org.springframework.validation.annotation.Validated
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RequestParam
+import org.springframework.web.servlet.mvc.support.RedirectAttributes
+
+@Controller
+@RequestMapping("/admin")
+@RequireGlobalRole([GlobalRole.SuperAdmin])
+@Validated
+class AdminVotersController(
+    private val defaultVoterStore: DefaultVoterStore,
+    private val dslContext: DSLContext,
+    private val userStore: UserStore,
+    private val voteStore: VoteStore,
+) {
+  private val log = perClassLogger()
+
+  @GetMapping("/voters")
+  fun getVoters(
+      @RequestParam email: String?,
+      model: Model,
+      redirectAttributes: RedirectAttributes,
+  ): String {
+    requirePermissions { updateDefaultVoters() }
+
+    val userIdForEmail = email?.let { userStore.fetchByEmail(it)?.userId }
+
+    val projects = getAcceleratorProjects()
+    val projectVoters =
+        projects.associateBy(
+            { it.projectId },
+            { project ->
+              voteStore
+                  .fetchAllVotes(project.projectId)
+                  .filter { it.phase === project.phase }
+                  .map { it.userId }
+                  .toSet()
+            })
+
+    val defaultVoters = defaultVoterStore.findAll()
+    val users =
+        defaultVoters
+            .union(projectVoters.values.reduce { acc, userIds -> acc.union(userIds) })
+            .map { userStore.fetchOneById(it) }
+            .associateBy { it.userId }
+
+    model.addAttribute("selectedEmail", email)
+    model.addAttribute("selectedUser", userIdForEmail)
+    model.addAttribute("defaultVoters", defaultVoters)
+    model.addAttribute("projects", projects)
+    model.addAttribute("projectVoters", projectVoters)
+    model.addAttribute("users", users)
+
+    return "/admin/voters"
+  }
+
+  @PostMapping("/voters/add")
+  fun addVoter(
+      @RequestParam email: String?,
+      @RequestParam userId: UserId,
+      @RequestParam projectId: ProjectId,
+      redirectAttributes: RedirectAttributes,
+  ): String {
+    requirePermissions { updateProjectVotes(projectId) }
+
+    val projects = getAcceleratorProjects().associateBy { it.projectId }
+    try {
+      voteStore.upsert(projectId, projects[projectId]!!.phase, userId)
+      redirectAttributes.successMessage = "Voter added."
+    } catch (e: Exception) {
+      log.error("Failed to add voter $userId to project $projectId", e)
+      redirectAttributes.failureMessage = "Failed to add voter: ${e.message}"
+    }
+
+    return redirectToVoters(email)
+  }
+
+  @PostMapping("/voters/remove")
+  fun removeVoter(
```

Comment:
Can this endpoint be combined with the bulk one? Seems like it could always take a list of user IDs, and for the single-user case the list will have one entry rather than multiple.

Same with the default voter remove endpoints.